python/openinference-instrumentation/tests/test_context_managers.py

```diff
@@ -23,56 +23,121 @@ def test_suppress_tracing() -> None:
+def test_using_user(user_id: str) -> None:
+    with using_user(user_id):
+        assert get_value(SpanAttributes.USER_ID) == user_id
     assert get_value(SpanAttributes.USER_ID) is None
 
 
-def test_using_metadata(metadata_fixture: Dict[str, Any]) -> None:
-    with using_metadata(metadata_fixture):
-        assert get_value(SpanAttributes.METADATA) == json.dumps(metadata_fixture, default=str)
+def test_using_metadata(metadata: Dict[str, Any]) -> None:
+    with using_metadata(metadata):
+        assert get_value(SpanAttributes.METADATA) == json.dumps(metadata, default=str)
     assert get_value(SpanAttributes.METADATA) is None
 
 
-def test_using_tags(tags_fixture: List[str]) -> None:
-    with using_tags(tags_fixture):
-        assert get_value(SpanAttributes.TAG_TAGS) == tags_fixture
+def test_using_tags(tags: List[str]) -> None:
+    with using_tags(tags):
+        assert get_value(SpanAttributes.TAG_TAGS) == tags
     assert get_value(SpanAttributes.TAG_TAGS) is None
 
 
-def test_using_attributes(metadata_fixture: Dict[str, Any], tags_fixture: List[str]) -> None:
+def test_using_attributes(
+    session_id: str, user_id: str, metadata: Dict[str, Any], tags: List[str]
+) -> None:
     with using_attributes(
-        session_id="test-session",
-        user_id="test-user",
-        metadata=metadata_fixture,
-        tags=tags_fixture,
+        session_id=session_id,
+        user_id=user_id,
+        metadata=metadata,
+        tags=tags,
     ):
-        assert get_value(SpanAttributes.SESSION_ID) == "test-session"
-        assert get_value(SpanAttributes.USER_ID) == "test-user"
-        assert get_value(SpanAttributes.METADATA) == json.dumps(metadata_fixture, default=str)
-        assert get_value(SpanAttributes.TAG_TAGS) == tags_fixture
+        assert get_value(SpanAttributes.SESSION_ID) == session_id
+        assert get_value(SpanAttributes.USER_ID) == user_id
+        assert get_value(SpanAttributes.METADATA) == json.dumps(metadata, default=str)
+        assert get_value(SpanAttributes.TAG_TAGS) == tags
+    assert get_value(SpanAttributes.SESSION_ID) is None
+    assert get_value(SpanAttributes.USER_ID) is None
+    assert get_value(SpanAttributes.METADATA) is None
+    assert get_value(SpanAttributes.TAG_TAGS) is None
+
+
+def test_using_session_decorator(session_id: str) -> None:
+    @using_session(session_id)
+    def f() -> None:
+        assert get_value(SpanAttributes.SESSION_ID) == session_id
+
+    f()
+    assert get_value(SpanAttributes.SESSION_ID) is None
+
+
+def test_using_user_decorator(user_id: str) -> None:
+    @using_user(user_id)
+    def f() -> None:
+        assert get_value(SpanAttributes.USER_ID) == user_id
+
+    f()
+    assert get_value(SpanAttributes.USER_ID) is None
+
+
+def test_using_metadata_decorator(metadata: Dict[str, Any]) -> None:
+    @using_metadata(metadata)
+    def f() -> None:
+        assert get_value(SpanAttributes.METADATA) == json.dumps(metadata, default=str)
```

Comment:
```suggestion
        assert get_value(SpanAttributes.METADATA) == json.dumps(metadata)
```

Nit: I think you can do without the `default` in these tests just because the metadata fixture is JSON serializable.

Reply:
Done